### DISCUSSION TITLE
Enhancement: Implement Access Control for Assistants in LibreChat

### DISCUSSION THREAD

**Discussion by Fakamoto** [danny-avila/LibreChat#2520] [Feature Requests & Suggestions]
### What features would you like to see added?

Implement enhanced access control mechanisms for uploaded documents and created assistants within our LibreChat instance. The goal is to enable private access for individual user uploads while also supporting public assistants accessible by all users.

### More details

Currently, all assistants are accessible by every user, all assistants are shared across the platform. This poses privacy and operational risks, especially when handling sensitive information. We propose the following enhancement:

Private and Public Assistants: Allow the creation of both private assistants, which are exclusive to the creator, and public assistants, which are available to all users.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
There are a couple of open issues related to this. Also, there is some access control already available: 
https://docs.librechat.ai/install/configuration/custom_config.html#assistants-endpoint-object-structure

https://github.com/danny-avila/LibreChat/issues/1926

Hope to address some of these concerns with introduction of v2 and beyond
